app/assets/javascripts/utilities/showUserAlertModal.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * Displays a general purpose user alert modal with a title, body text, and confirmation button.
+ *
+ * @function showUserAlertModal
+ * @param {string} title The title/heading text to be displayed
+ * @param {string} text The body text to be displayed
+ * @param {string} confirm_text Text of the confirmation button
+ * 
+ * @example 
+ * showUserAlertModal('Warning', 'You must wait', 'OK', '/faq/why-must-i-wait', 'Why must I wait?');
+ */
+function showUserAlertModal(title, text, confirm_text) {
+  let modalDiv = buildModalDiv(title, text, confirm_text);
+  toggleUserAlertModal();
+}
+/**
+ * Displays a user rate limit alert modal letting the user know what they did that exceeded a rate limit, 
+ * and gives them links to explain why they must wait
+ *
+ * @function showUserAlertModal
+ * @param {string} action_text Description of the action taken by the user
+ * @param {string} next_action_text Description of the next action that can be taken
+ *  
+ * @example 
+ * showRateLimitModal('Made a comment', 'comment again');
+ */
+function showRateLimitModal(action_text, next_action_text) {
+  let rateLimitText = buildRateLimitText(action_text, next_action_text);
+  let rateLimitLink = "/faq";
+  showUserAlertModal('Wait a Moment...', rateLimitText, 'Got it', rateLimitLink, "Why do I have to wait?")
+}
+
+/**
+ * HTML ID for modal DOM node
+ * 
+ * @private
+ * @constant modalId * 
+ * @type {string}
+ */
+const modalId = 'user-alert-modal';
+
+/**
+ * HTML template for modal
+ * 
+ * @private
+ * @function modalHTML
+ * 
+ * @param {string} title The title/heading text to be displayed
+ * @param {string} text The body text to be displayed
+ * @param {string} confirm_text Text of the confirmation button
+ * 
+ * @returns {string} HTML for the modal
+ */
+const modalHTML = (title, text, confirm_text) => `<div id="${modalId}" data-testid="modal-container" class="crayons-modal hidden">
```

Comment:
nitpick (non-blocking): Consider naming this `getModalHtml` or `buildModalHtml`.